packages/app/src/cli/models/app/loader.ts

```diff
@@ -247,30 +311,26 @@ class AppLoader {
         UIExtensionConfigurationSupportedSchema,
         configurationPath,
       )
+
       const configuration = {
         ...configurationSupported,
         type: mapUIExternalExtensionTypeToUIExtensionType(configurationSupported.type),
       }
 
-      const entrySourceFilePaths = await path.glob(path.join(directory, 'src', '*.+(ts|js|tsx|jsx)'))
+      let entrySourceFilePath: string | undefined
 
-      if (!entrySourceFilePaths[0]) {
-        // TODO: Previosuly this was abortOrReport, but aborting guarantees types safety
-        // In what circumstance would we want this process to continue if there are no src files?
-        // There being no src files seems like a terminal problem.
-        throw new error.Abort(
-          output.content`Couldn't find any js, jsx, ts or tsx files in the directories ${output.token.path(
-            directory,
-          )} or ${output.token.path(path.join(directory, 'src'))}`,
-        )
+      if (await isLegacyUIExtension(configuration)) {
+        entrySourceFilePath = await this.getLegacyEntrySourceFilePath(directory)
+      } else {
+        await this.validateUIExtensionPointConfig(directory, configuration.extensionPoints as NewExtensionPointType)
```

Comment:
I spent a bit of time trying to avoid as here. In the end I decided it was not worth the effort since:

The plan is this code will never get merged into master.
After Isaac's refactor, and migrating existing extensions to the ui_extension type, this should be much simpler